avocado/core/test.py

```diff
@@ -105,6 +105,11 @@ def record_and_warn(*args, **kwargs):
 
         # Replace '/' with '_' to avoid splitting name into multiple dirs
         safe_tagged_name = astring.string_to_safe_path(self.tagged_name)
+        test_log_dir = os.path.join(base_logdir, safe_tagged_name)
+        if os.path.isdir(test_log_dir):
+            msg = ('Test log dir "%s" exists prior to Avocado creating '
+                   'it' % test_log_dir)
+            raise exceptions.TestSetupFail(msg)
```

Comment:
The use case is that, without it, Avocado will happily reuse an existing test log dir. It shouldn't be allowed so it's a explicit safeguard.

Line 322 updates the tagged name, but between then and the directory creation, unexpected things can happen.